### ISSUE TITLE
arm-linux-androideabi build failures

### ISSUE THREAD

**Issue by posborne** [nix-rust/nix#298]
Using the new docker test infrastructure...

```
$ RUST_VERSIONS=stable RUST_TARGETS=arm-linux-androideabi ./ci/run-all.sh
...
     Running `rustc /root/.multirust/toolchains/1.7.0/cargo/registry/src/github.com-88ac128001ac3a9a/advapi32-sys-0.1.2/build.rs --crate-name build_script_build --crate-type bin -g --out-dir /build/debug/build/advapi32-sys-911258561df3b2a9 --emit=dep-info,link -L dependency=/build/debug/deps -L dependency=/build/debug/deps --extern build=/build/debug/deps/libbuild-7bc4b8a4c9d61577.rlib --cap-lints allow`
/source/src/sys/termios.rs:45:13: 45:38 error: unresolved import `libc::funcs::bsd44::ioctl`. Could not find `funcs` in `libc` [E0432]
/source/src/sys/termios.rs:45         use libc::funcs::bsd44::ioctl;
                                          ^~~~~~~~~~~~~~~~~~~~~~~~~
/source/src/sys/termios.rs:45:13: 45:38 help: run `rustc --explain E0432` to see a detailed explanation
/source/src/sys/socket/addr.rs:550:29: 550:40 error: unresolved import `libc::sockaddr_nl`. There is no `sockaddr_nl` in `libc`. Did you mean to use `sockaddr_ll`? [E0432]
/source/src/sys/socket/addr.rs:550     use libc::{sa_family_t, sockaddr_nl};
                                                               ^~~~~~~~~~~
/source/src/sys/socket/addr.rs:550:29: 550:40 help: run `rustc --explain E0432` to see a detailed explanation
```

**Comment by posborne** [comment]
`sockaddr_nl` should be resolved by https://github.com/rust-lang/libc/pull/225

**Comment by Susurrus** [comment]
This was addressed sometime before #639 landed as this target is now checked during CI.
